packages/nuxt/src/app/composables/router.ts

```diff
@@ -97,8 +98,15 @@ export const navigateTo = (to: RouteLocationRaw | undefined | null, options?: Na
   if (process.server) {
     const nuxtApp = useNuxtApp()
     if (nuxtApp.ssrContext && nuxtApp.ssrContext.event) {
+      // Let vue-router handle internal redirects within middleware
+      // to prevent the navigation happening after response is sent
+      if (isProcessingMiddleware() && !isExternal) {
+        setResponseStatus(options?.redirectCode || 302)
+        return to
```

Comment:
Would it properly work with prerendering?

Reply:
It should behave the same as the current behavior.

Reply:
(The below code will still be run, just after the middleware finishes processing.)

Reply:
It is returning but i trust you it will 👍🏼

Reply:
This will call `navigateTo` again:

https://github.com/nuxt/nuxt.js/blob/83a8e1378e565784f3ae18026ca7c36751f384e9/packages/nuxt/src/pages/runtime/plugins/router.ts#L181-L185